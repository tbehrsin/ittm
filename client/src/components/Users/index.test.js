import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { act, render } from '@testing-library/react';
import { Users, GET_USERS_QUERY } from './index';
import mockUsers from '../../../../mock-object.json';

const immediate = async () => new Promise(resolve => setImmediate(resolve));

mockUsers.data.items.forEach(item => {
  item.id = item.id.toString();
  item.role.id = item.role.id.toString();
});

const mocks = [
  {
    request: {
      query: GET_USERS_QUERY,
    },
    result: {
      data: {
        users: mockUsers?.data?.items,
      },
    },
  },
];

describe('Users', () => {
  test('renders without crashing', async () => {
    const { asFragment } = render(
      <MockedProvider mocks={mocks}>
        <Users />
      </MockedProvider>
    );

    expect(asFragment()).toMatchSnapshot();

    await act(immediate);

    expect(asFragment()).toMatchSnapshot();
  });
});
