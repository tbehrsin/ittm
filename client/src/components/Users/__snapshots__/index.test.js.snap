// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders without crashing 1`] = `
<DocumentFragment>
  <table
    class="container"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Email
        </th>
        <th>
          Region
        </th>
        <th>
          Blocked
        </th>
        <th>
          Status
        </th>
        <th>
          Created
        </th>
        <th>
          Role ID
        </th>
        <th>
          Role Name
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;

exports[`Users renders without crashing 2`] = `
<DocumentFragment>
  <table
    class="container"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Email
        </th>
        <th>
          Region
        </th>
        <th>
          Blocked
        </th>
        <th>
          Status
        </th>
        <th>
          Created
        </th>
        <th>
          Role ID
        </th>
        <th>
          Role Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="container"
      >
        <td>
          123
        </td>
        <td>
          support_admin@example.com
        </td>
        <td>
          UK
        </td>
        <td>
          false
        </td>
        <td>
          ACTIVE
        </td>
        <td>
          Monday 6th April 2020
        </td>
        <td>
          1
        </td>
        <td>
          test role
        </td>
      </tr>
      <tr
        class="container"
      >
        <td>
          456
        </td>
        <td>
          yolo.swaggins@example.com
        </td>
        <td>
          UK
        </td>
        <td>
          false
        </td>
        <td>
          PENDING
        </td>
        <td>
          Monday 6th April 2020
        </td>
        <td>
          1
        </td>
        <td>
          test role
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
